lib/rubocop/cop/rspec/message_chain.rb

```diff
@@ -0,0 +1,35 @@
+module RuboCop
+  module Cop
+    module RSpec
+      # Check that chains of messages are not being stubbed.
+      #
+      # Prefer instance doubles over stubbing any instance of a class
+      #
+      # @example
+      #   # bad
+      #   allow(foo).to receive_message_chain(:bar, :baz).and_return(42)
+      #
+      #   # better
+      #   thing = Thing.new(baz: 42)
+      #   allow(foo).to receive(bar: thing)
```

Comment:
Sometimes I'll give an example like this:

```
# bad
...

# okay
... (intermediary step)

# good
... (ideal outcome)
```

where the "okay" step is something minimally better (here that might be `allow(foo).to receive(bar).and_return(instance_double(Thing, baz: 42))`) while "good" might be an example that actually shows an improved design under test.

WDYT?